entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityService.kt

```diff
@@ -51,34 +51,28 @@ import java.net.URI
 class EntityService(
     private val neo4jRepository: Neo4jRepository,
     private val entityRepository: EntityRepository,
-    private val entitiesGraphBuilder: EntitiesGraphBuilder,
+    private val partialEntityRepository: PartialEntityRepository,
     private val propertyRepository: PropertyRepository,
     private val applicationEventPublisher: ApplicationEventPublisher
 ) {
     private val logger = LoggerFactory.getLogger(javaClass)
 
     @Transactional
-    fun createEntity(ngsiLdEntity: NgsiLdEntity): Entity {
-        if (exists(ngsiLdEntity.id)) {
+    fun createEntity(ngsiLdEntity: NgsiLdEntity): URI {
+        if (exists(ngsiLdEntity.id))
             throw AlreadyExistsException("Already Exists")
-        }
-
-        val (_, invalidRelationsErrors) = entitiesGraphBuilder.build(listOf(ngsiLdEntity))
-        if (invalidRelationsErrors.isNotEmpty()) {
-            // we have just one entity in validation, so can't have more than one relations error
-            // in its current version, it will only contain one error but it will be fixed with validation v2
-            val inErrorRelationships = invalidRelationsErrors[0].error.joinToString(",")
-            throw BadRequestDataException("Entity ${ngsiLdEntity.id} targets unknown entities: $inErrorRelationships")
-        }
 
+        val isPartial = isPartial(ngsiLdEntity.id)
         val rawEntity =
             Entity(id = ngsiLdEntity.id, type = listOf(ngsiLdEntity.type), contexts = ngsiLdEntity.contexts)
-        val entity = entityRepository.save(rawEntity)
+        entityRepository.save(rawEntity)
+        if (isPartial)
```

Comment:
you could use isPartial(ngsiLdEntity.id) in the if statement since the val isPartial is not used elsewhere